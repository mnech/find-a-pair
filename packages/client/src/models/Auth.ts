export enum UserStatusTypes {
  BEGIN = 'BEGIN',
  FAILURE = 'FAILURE',
  SUCCESS = 'SUCCESS',
}

export interface OAuthData {
  code: string;
  redirect_uri: string;
}
