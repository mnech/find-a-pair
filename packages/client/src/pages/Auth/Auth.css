.login {
    height: min-content;
    padding: 20px;
    border-radius: 12px;
    background: #ffffff;
}