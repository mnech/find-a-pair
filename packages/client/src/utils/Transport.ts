import axios from 'axios';
import { Methods } from '../api/consts';

export type TRequest = {
  method?: Methods
  url: string
  data?: unknown
};

export default class Transport {
  protected axios = axios;

  constructor() {
    this.axios.defaults.withCredentials = true;
  }

  public request({ method = Methods.Get, url, data }: TRequest) {
    return this.axios({ method, url, data });
  }
}
