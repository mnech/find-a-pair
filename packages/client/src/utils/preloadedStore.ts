import {
  limitPage,
  paginationMin,
  ratingFieldName,
} from '../models/Leaderboard';

export const preloadedState = {
  leaderboard: {
    //пока заглушка
    data: [
      {
        name: 'User',
        score: 1,
      },
      {
        name: 'Jane',
        score: 2,
      },
      {
        name: 'James',
        score: 7,
      },
      {
        name: 'Luis',
        score: 10,
      },
    ],
    request: {
      ratingFieldName: ratingFieldName,
      cursor: paginationMin,
      limit: limitPage,
    },
  },
};
