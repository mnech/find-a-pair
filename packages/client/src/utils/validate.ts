import { Regexp } from '../consts';

export const regexpTest = (type: keyof typeof Regexp, str: string) => {
  return Regexp[type].test(str);
};
