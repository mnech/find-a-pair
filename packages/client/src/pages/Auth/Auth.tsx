import React, { useState } from 'react';
import { Form, Button } from 'react-bootstrap';
import AuthController from '../../controllers/AuthController';
import { useSelector } from 'react-redux';
import { routes } from '../../models/App';
import { Link } from 'react-router-dom';
import { RootState } from '../../store';
import ContinueWithYandex from '../../components/continueWithYandex/ContinueWithYandex';

export const Auth = () => {
  const [login, setLogin] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const error = useSelector((state: RootState) => state.auth.error);
  const handleChangeLogin = (event: React.ChangeEvent<HTMLInputElement>) => {
    setLogin(event.currentTarget.value);
  };
  const handleChangePasword = (event: React.ChangeEvent<HTMLInputElement>) => {
    setPassword(event.currentTarget.value);
  };

  const handleClickSubmit = async (event: React.FormEvent) => {
    event.preventDefault();
    await AuthController.signin({ login, password });
  };

  return (
    <div className="container-sm mt-5" style={{ padding: '10em 0' }}>
      {error && (
        <div className="alert alert-danger col-lg-4 offset-lg-4" role="alert">
          {error}
        </div>
      )}
      <Form
        className="row d-flex justify-content-center col-lg-4 offset-lg-4 shadow-sm p-5 rounded needs-validation"
        onSubmit={handleClickSubmit}
      >
        <div className="form-outline">
          <input
            type="text"
            id="login"
            className="form-control form-control-lg"
            required
            onChange={handleChangeLogin}
            value={login}
          />
          <label className="form-label" htmlFor="login">
            Логин
          </label>
        </div>
        <div className="form-outline">
          <input
            type="password"
            id="password"
            className="form-control form-control-lg"
            onChange={handleChangePasword}
            value={password}
            required
          />
          <label className="form-label" htmlFor="password">
            Пароль
          </label>
        </div>
        <Button type="submit" className="btn btn-primary w-50 mb-2">
          Войти
        </Button>
        <ContinueWithYandex />
        <div className="text-center">
          <Link to={routes.signUp} className="link-primary">
            Нет аккаунта?
          </Link>
        </div>
      </Form>
    </div>
  );
};
