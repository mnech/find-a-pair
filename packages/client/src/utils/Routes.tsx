import { ErrorBoundary } from './ErrorBoundary';
import { Auth } from '../pages/Auth/Auth';
import { Leaderboard } from '../pages/Leaderboard/Leaderboard';
import Game from '../pages/Game/Game';
import Forum from '../pages/Forum/Forum';
import Profile from '../pages/Profile/Profile';
import { StartGameScreen } from '../pages/StartGameScreen/StartGameScreen';
import Signup from '../pages/SignUp/SignUp';
import { routes } from '../models/App';
import { EndGameScreen } from '../pages/EndGameScreen/EndGameScreen';
export const staticRoutes = [
  {
    path: routes.auth,
    errorElement: <ErrorBoundary />,
    children: [
      {
        index: true,
        path: routes.auth,
        element: <Auth />,
      },
      {
        path: routes.profile,
        element: <Profile />,
      },
      {
        path: routes.signUp,
        element: <Signup />,
      },
      {
        path: routes.leaderboard,
        element: <Leaderboard />,
      },
      {
        path: routes.forum,
        element: <Forum />,
      },
      {
        path: routes.game,
        element: <Game />,
      },
      {
        path: routes.startGame,
        element: <StartGameScreen />,
      },
      {
        path: routes.endGame,
        element: <EndGameScreen />,
      },
    ],
  },
];
