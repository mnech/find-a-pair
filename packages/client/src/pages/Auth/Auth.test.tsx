import { renderWithProviders } from '../../utils/configureStore';
import { Auth } from './Auth';

test('Auth renders correctly', () => {
  const { asFragment } = renderWithProviders(<Auth />);
  expect(asFragment()).toMatchSnapshot();
});
