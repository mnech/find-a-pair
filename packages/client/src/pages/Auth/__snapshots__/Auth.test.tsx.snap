// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container-sm mt-5"
    style="padding: 10em 0px;"
  >
    <form
      class="row d-flex justify-content-center col-lg-4 offset-lg-4 shadow-sm p-5 rounded needs-validation"
    >
      <div
        class="form-outline"
      >
        <input
          class="form-control form-control-lg"
          id="login"
          required=""
          type="text"
          value=""
        />
        <label
          class="form-label"
          for="login"
        >
          Логин
        </label>
      </div>
      <div
        class="form-outline"
      >
        <input
          class="form-control form-control-lg"
          id="password"
          required=""
          type="password"
          value=""
        />
        <label
          class="form-label"
          for="password"
        >
          Пароль
        </label>
      </div>
      <button
        class="btn btn-primary w-50 btn btn-primary"
        type="submit"
      >
        Войти
      </button>
      <div
        class="text-center"
      >
        <button
          class="btn btn-link"
          type="button"
        >
          Нет аккаунта?
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
